configuration/download_cache.rst

```diff
@@ -0,0 +1,92 @@
+.. _download_cache:
+
+Download cache
+==============
+
+.. warning::
+
+    This is an **experimental** feature subject to breaking changes in future releases.
+
+.. warning::
+
+    This is an **advanced** feature, use it only if you fully understand this document. Do not use it if you are mixing api V1 (without revisions)
+    and api V2 (with revisions) clients.
+
+
+
+Conan implements a shared download cache that can be used to reduce the time needed to populate the Conan package cache
+with commands like :command:`install`, :command:`create`.
+
+This cache is purely an optimization mechanism. It is completely different to the :ref:`Conan package cache <custom_cache>`, (typically the ``<userhome>/.conan`` folder).
+It is not related to the ``short_paths`` mechanism for long path in Windows, nor to the ``short_paths`` cache folder. The cache will contain a copy
+of the artifacts, it is not a new location of files. Those files will still be copied to the Conan package cache, which will not change anything,
+its behavior, layout or location of any file.
+
+This cache (whose path can be configured in the *conan.conf* file) will store the following items:
+
+- All files that are downloaded from a Conan server (conan_server, Artifactory), both in the api V1 (without revisions) and V2 (with revisions).
+  This includes files like *conanfile.py*, but also the zipped artifacts like *conan_package.tgz* or *conan_sources.tgz*.
+- The downloads done by users with the ``tools.download()`` or ``tools.get()`` helpers, as long as they provide a checksum (md5, sha1, etc.). If
+  a checksum is not provided, even if the download cache is enabled, the download will be always executed and the files will not be cached.
+
+The cache uses the URL of the download, appending the checksum when provided. In the api V2, it is not necessary to append the checksum, because
+the URL itself already encodes the recipe revision and/or the package revisions, which are already checksums of the recipe and package respectively.
+
+.. warning::
+
+    The download cache will not be able to correctly cache artifacts with revisions enabled if those artifacts are created and uploaded repeatedly
+    in a client without revisions, because that will keep overwriting the revision "0".
```

Comment:
Same issue if the user changes the proxy, maybe we can mention it here. Those changes won't invalidate the cache

Reply:
Yes, lets add it.